simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -143,9 +145,16 @@ SwitchMmu::SetEgressLosslessPool(uint64_t b) {
 void
 SwitchMmu::SetReserved(uint64_t b, uint32_t port, uint32_t q, std::string inout) {
 	if (inout == "ingress") {
+		if (totalIngressReserved >= reserveIngress[port][q])
+			totalIngressReserved -= reserveIngress[port][q];
+		else
+			totalIngressReserved = 0;
 		reserveIngress[port][q] = b;
+		totalIngressReserved += reserveIngress[port][q];
 	}
 	else if (inout == "egress") {
+		std::cout << "setting headroom for egress is not supported. Exiting..!" << std::endl;
```

Comment:
This should be "minimum guaranteed buffer" rather than "headroom".

Reply:
changed.